src/main/java/de/caritas/cob/userservice/api/service/emailsupplier/TenantDataSupplier.java

```diff
@@ -0,0 +1,53 @@
+package de.caritas.cob.userservice.api.service.emailsupplier;
+
+import de.caritas.cob.userservice.api.tenant.TenantContext;
+import de.caritas.cob.userservice.mailservice.generated.web.model.TemplateDataDTO;
+import de.caritas.cob.userservice.tenantservice.generated.web.TenantControllerApi;
+import de.caritas.cob.userservice.tenantservice.generated.web.model.RestrictedTenantDTO;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class TenantDataSupplier {
+
+  private final TenantControllerApi tenantControllerApi;
+
+  @Value("${app.base.url}")
+  private String applicationBaseUrl;
+
+  public List<TemplateDataDTO> getTemplateAttributes() {
+    var subdomain = TenantContext.getCurrentTenantData().getSubdomain();
+    RestrictedTenantDTO tenantData = tenantControllerApi
+        .getRestrictedTenantDataBySubdomain(subdomain);
+
+    List<TemplateDataDTO> templateAttributes = new ArrayList<>();
+    templateAttributes.add(new TemplateDataDTO().key("tenant_name").value(tenantData.getName()));
```

Comment:
template keys could be extracted to static constants for better readability

Reply:
are now contained in readable method names, and used only inside this methods